src/conditions/remap.rs

```diff
@@ -57,7 +57,10 @@ impl Remap {
         // program wants to mutate its events.
         //
         // see: https://github.com/timberio/vector/issues/4744
-        Runtime::default().execute(&mut event.clone(), &self.program)
+        match event {
+            Event::Log(event) => Runtime::default().execute(&mut event.clone(), &self.program),
+            Event::Metric(event) => Runtime::default().execute(&mut event.clone(), &self.program),
+        }
```

Comment:
Not a big deal, but we can move some duplication out of this match by casing `event` to an `dyn Object` inside it, and then leave the rest of the line out of the match.

Reply:
It would then need boxing etc.. I felt it wasn't worth the extra overhead to save the bit of code duplication.

Reply:
Ah yeah, I see. I guess you could use generics (you can define a function within this function). But it's fine the way it is as well 😄

Reply:
That's a nice idea, but I can't get it to work..

```rust
        let execute = |object| Runtime::default().execute(object, &self.program);

        match event {
            Event::Log(event) => execute(&mut event.clone()),
            Event::Metric(event) => execute(&mut event.clone()),
        }
```
`execute` is monomorphized and so can't handle both paths. Casting to `dyn remap::Object` then falls down as it is unsized..

I could move execute into a top level function in the struct, but the benefits of doing so then rapidly diminish!

Reply:
Yeah, you'd have to use a function, as closure's can't take generic parameters. You can still define the function within this function, but I agree, the benefits aren't as high anymore.